src/kinesis.js

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash-firecloud';
+import aws from 'aws-sdk';
+
+// see https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html
+export let limits = {
+  batchByteSize: 5 * 1024 * 1024,
+  batchRecord: 500,
+  recordByteSize: 1024 * 1024
+};
+
+let _putRecordBatches = async function({
+  kinesis,
+  recordBatches
+}) {
+  let processedCount = 0;
+
+  // eslint-disable-next-line fp/no-loops, better/no-fors
+  for (let recordBatch of recordBatches) {
+    delete recordBatch.byteSize;
+    await kinesis.putRecordBatch(recordBatch).promise();
+    processedCount = processedCount + recordBatch.Records.length;
+  }
+
+  return processedCount;
+};
+
+export let putRecords = async function({
+  ExplicitHashKey,
+  PartitionKey,
+  StreamName,
+  ctx,
+  kinesis = new aws.Kinesis(),
+  records
+}) {
+  let recordBatches = [];
+  let recordBatch = {
+    StreamName,
+    Records: [],
+    byteSize: 0
+  };
+
+  let toProcessCount = records.length;
+
+  _.forEach(records, function(record) {
+    let Data = JSON.stringify(record);
+    let dataLength = Buffer.byteLength(JSON.stringify({
+      Data: record,
+      ExplicitHashKey,
+      PartitionKey
+    }));
+
+    if (dataLength > limits.recordByteSize) {
+      ctx.log.error(`Skipping record larger than ${limits.recordByteSize / 1024} KB: \
```

Comment:
I'm questioning now this approach where a lib function almost silently fails to send records. I know we have it in the firehose too, but still. I would say it's up to the app to decide how to deal with such records.

Should we maybe just expose a helper function that checks a record size and allows a client to decide what to do with non-conforming records?

Reply:
Sounds good! But as part of another PR